types/signed_header.go

```diff
@@ -102,18 +104,26 @@ func (sh *SignedHeader) ValidateBasic() error {
 	}
 
 	// Check that the proposer address in the signed header matches the proposer address in the validator set
-	if !bytes.Equal(sh.ProposerAddress, sh.Signer.Address) {
+	signerAddress, err := sh.Signer.GetAddress()
+	if err != nil {
+		return err
+	}
+	if !bytes.Equal(sh.ProposerAddress, signerAddress) {
 		return ErrProposerAddressMismatch
 	}
 
 	signature := sh.Signature
 
-	vote, err := sh.Header.Vote()
+	bz, err := sh.Header.MarshalBinary()
 	if err != nil {
 		return err
 	}
 
-	verified, err := sh.Signer.PubKey.Verify(vote, signature)
+	pubkey, err := sh.Signer.GetPublic()
```

Comment:
this is incorrect as how will non signing nodes have the public key here?

Reply:
Pubkey of the proposer is included in the signed header afaik